src/main/java/com/fryrank/dal/UserMetadataDALImpl.java

```diff
@@ -0,0 +1,41 @@
+package com.fryrank.dal;
+
+import com.fryrank.model.GetAllReviewsOutput;
+import com.fryrank.model.GetUserMetadataOutput;
+import com.fryrank.model.UserMetadata;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.core.FindAndReplaceOptions;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.stereotype.Repository;
+
+
+import java.util.List;
+
+import static com.fryrank.Constants.ACCOUNT_ID_KEY;
+
+@Repository
+public class UserMetadataDALImpl implements UserMetadataDAL {
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    @Override
+    public GetUserMetadataOutput getUserMetadataForAccountId(String accountId) {
+        final Query query = new Query();
+        final Criteria equalToUserIdCriteria = Criteria.where(ACCOUNT_ID_KEY).is(accountId);
+        query.addCriteria(equalToUserIdCriteria);
+        final List<UserMetadata> userMetadata = mongoTemplate.find(query, UserMetadata.class);
+
```

Comment:
Can we insert logic here to check for one element returned and then return that one element instead of the list?

Reply:
Done in [e281f82](https://github.com/NickPriv/FryRankBackend/pull/75/commits/e281f825a8834d7e490af094d3f5f49d6a0f9de1).